src/Sylius/Bundle/ResourceBundle/Form/DataTransformer/ResourceToIdentifierTransformer.php

```diff
@@ -49,7 +49,7 @@ public function __construct(RepositoryInterface $repository, $identifier = 'id')
     public function transform($value)
     {
         if (null === $value) {
-            return '';
+            return null;
```

Comment:
I don't like it also, but remember that Symfony recommended that.

From `DataTransformerInterface`:
```
     * This method must be able to deal with empty values. Usually this will
     * be NULL, but depending on your implementation other empty values are
     * possible as well (such as empty strings). The reasoning behind this is
     * that value transformers must be chainable. If the transform() method
     * of the first value transformer outputs NULL, the second value transformer
     * must be able to process that value.
     *
     * By convention, transform() should return an empty string if NULL is
     * passed.
```

By convention it should return empty string, but not sure why 😄